client/snap_op.go

```
@@ -197,8 +204,13 @@ func (client *Client) doMultiSnapActionFull(actionName string, snaps []string, o
 		Snaps:  snaps,
 	}
 	if options != nil {
+		if options.Dangerous {
+			return nil, "", ErrDangerousNotApplicable
```

This introduces a change in the behavior of snap install, previously `snap install foo bar --dangerous` would just silently ignore the dangerous flag what means that people could have scripted such operations and they did not fail. Now we actively return `error: dangerous option only meaningful when installing from a local file`. Perhaps we should continue ignoring most flags here, and tweak the failing unit test that the flag (and maybe some other flags) is indeed ignored.

---

we might want to consider this change, probably separately from the rest of the feature though. It's unfortunate that we have simply ignored all flags so far for the many case.

---

I've changed it so it does not error out now and added a TODO